sdk/trace-shaded-deps/src/main/java/io/opentelemetry/sdk/trace/internal/JcTools.java

```diff
@@ -64,21 +64,23 @@ public static long capacity(Queue<?> queue) {
    * @throws IllegalArgumentException if maxExportBatchSize is negative
    */
   @SuppressWarnings("unchecked")
-  public static <T> void drain(Queue<T> queue, int limit, Consumer<T> consumer) {
+  public static <T> int drain(Queue<T> queue, int limit, Consumer<T> consumer) {
```

Comment:
Also it would be easier to optimize if `MessagePassingQueue.Consumer` was used as consumer parameter type, so we don't need to rely on C2 optimizations